src/sentry/grouping/strategies/newstyle.py

```diff
@@ -279,6 +299,38 @@ def frame(frame, event, context, **meta):
     if context_line_component is not None:
         values.append(context_line_component)
 
+    if (
+        context["discard_native_filename"]
+        and get_behavior_family_for_platform(platform) == "native"
+        and function_component.contributes
+        and filename_component.contributes
+    ):
+        # In native, function names usually describe a full namespace. Adding
+        # the filename there just brings extra instability into grouping.
+        filename_component.update(
+            contributes=False, hint="discarded native filename for grouping stability"
+        )
+
+    if context["use_package_fallback"] and frame.package:
+        # If function did not symbolicate properly and we also have no filename, use package as fallback.
+        package_component = get_package_component(package=frame.package, platform=platform)
+        if package_component.contributes:
+            use_package_component = all(not component.contributes for component in values)
+
+            if use_package_component:
+                package_component.update(
+                    hint="used as fallback because function name is not available"
```

Comment:
Just trying to understand the code here: should this hint read "used as fallback because module / filename / function is not available"?

Reply:
the package fallback is specifically for native where we don't receive a module and intentionally discard the filename anyway

other platforms don't set package afaik